### PR TITLE
Task1 gcd lcm

### PR DESCRIPTION
**Pull request by mikabele**

### CODE REVIEW

File: src/test/scala/CalculatorTest.scala [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import org.scalatest.funsuite.AnyFunSuite
+
+class CalculatorTest extends AnyFunSuite{
+  test("Test gcd when a greater than b"){
```
Comment (itechart-scala-lab) [inline src/test/scala/CalculatorTest.scala:4]: add corner case for lcm diviison / 0

File: src/main/scala/Calculator.scala [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import scala.annotation.tailrec
+
+object Calculator {
+  def lcm(a: Int, b: Int): Int = {
```
Comment (itechart-scala-lab) [inline src/main/scala/Calculator.scala:4]:
```suggestion
  def lcm(a: Int, b: Int): Option[Int] = {
```

File: src/main/scala/Calculator.scala [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import scala.annotation.tailrec
+
+object Calculator {
+  def lcm(a: Int, b: Int): Option[Int] = {
+    gcd(a, b) match {
+      case 0 => None
+      case calcGCD => Some(Math.abs(a / calcGCD * b))
```
Comment (gleb-streltsov-by) [inline src/main/scala/Calculator.scala:7]: Maybe just `gcd` instead `calcGCD`?

File: src/main/scala/Calculator.scala [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    }
+  }
+
+  @tailrec
+  def gcd(a: Int, b: Int): Int = {
+    if (b == 0)
+      a
+    else
+      gcd(b, a % b)
```
Comment (gleb-streltsov-by) [inline src/main/scala/Calculator.scala:16]: With short lines you can make it even like this:

```suggestion
    if (b == 0) a
    else gcd(b, a % b)
```

File: src/test/scala/CalculatorTest.scala [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+  test("Test lcm when a greater than b - successful path"){
+    assert(Calculator.lcm(10,25).getOrElse(0)==50)
+  }
+  test("Test lcm when b or a equals to 0"){
+    assert(Calculator.lcm(25,0).getOrElse(-1)==0)
+  }
+  test("Test lcm when a and b equal to 0"){
+    assert(Calculator.lcm(0,0).isEmpty)
+  }
+}
```
Comment (gleb-streltsov-by) [inline src/test/scala/CalculatorTest.scala:19]: Tests with negative values is one of corner cases. Ideally, it would be nice to consider as well.

**Review by gleb-streltsov-by — APPROVED** [review]
